### ISSUE TITLE
About SimCLR model 

### ISSUE THREAD

**Issue by szc19990412** [binli123/dsmil-wsi#7]
Great Work! And I want to know how long it took you to train the SimCLR model? Can you share the trained ResNet18 model.pth file?

**Comment by binli123** [comment]
It took two weeks to train for the 20x magnification on a 6-GPU (1080Ti) server due to the large number of patches, with a batch size of 512. Bigger batch size and longer training time will lead to better representations. 
For TCGA Lung Cancer dataset, training without considering multiple instance learning but rather treating each patch as if it has the same label as the slide could also lead to a good embedded due to that large portions of a slide contain tumors. The model would achieve ~85% accuracy in testing by simply averaging the scores of the patches and determine a threshold.  With this embedder, you can then project each patch into a vector and consider multiple instance learning, which increases the classification accuracy to ~95%.
Using embedder learning with SimCLR could lead to similar accuracy.
A trained model weights file can be downloaded using download.py. The URL is https://uwmadison.box.com/shared/static/grxja488s4i07h9wo3tm4sj6t4nqtz0b.zip

**Comment by Amcky** [comment]
Can you share the trained ResNet18 model.pth file?(Camelyon16)

**Comment by Dadatata-JZ** [comment]
@binli123 Regarding the embedder.pth you posted (thanks), was it trained on one gpu or 6? since when I loaded it to the ResNetSimCLR, I got complaints about missing keys.

Additionally, to clarify, when you mentioned that "It took two weeks to train for the 20x magnification on a 6-GPU (1080Ti) server ", the model was trained on ResNet 18 without any pre-trained weights right? no even the ImageNet pretrained to init. 

Thanks in advance

**Comment by binli123** [comment]
> @binli123 Regarding the embedder.pth you posted (thanks), was it trained on one gpu or 6? since when I loaded it to the ResNetSimCLR, I got complaints about missing keys.
> 
> Additionally, to clarify, when you mentioned that "It took two weeks to train for the 20x magnification on a 6-GPU (1080Ti) server ", the model was trained on ResNet 18 without any pre-trained weights right? no even the ImageNet pretrained to init.
> 
> Thanks in advance

Yes, not even with ImageNet pretrained. Plus, instance normalization is used instead of batchnorm
